File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s)-1
        while (j>i):
            while (s[i].isalnum()== False and j>i):
                i+=1
            while(s[j].isalnum()==False and j>i):
                j-=1
            if(s[i].lower() != s[j].lower()):
                return False
            j-=1
            i+=1
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i =0,j = s.size()-1;
        while(i<j){
            while(i< s.size() and iswalnum(s[i])==false)++i;
            while(j>=0 and iswalnum(s[j])==false)--j;
            if(j<=i) continue;
            if(tolower(s[i])!=tolower(s[j]))
                return false;
            ++i;--j;
        }
        return true;
    }
};